Mimir/Controllers/AgentController.cs

```diff
@@ -1,32 +1,77 @@
+using Lib9c;
 using Libplanet.Crypto;
+using Libplanet.Types.Assets;
 using Microsoft.AspNetCore.Mvc;
 using Mimir.Models.Agent;
+using Mimir.Models.Assets;
 using Mimir.Services;
 using Mimir.Util;
 
 namespace Mimir.Controllers;
 
 [ApiController]
-[Route("{network}/agent")]
+[Route("{network}/agent/{address}")]
 public class AgentController : ControllerBase
 {
-    [HttpGet("{agentAddress}/avatars")]
+#pragma warning disable CS0618 // Type or member is obsolete
+    private readonly Currency _ncg = Currency.Legacy(
+        "NCG",
+        2,
+        new Address("0x47D082a115c63E7b58B1532d20E631538eaFADde"));
+#pragma warning restore CS0618 // Type or member is obsolete
+
+    [HttpGet("balances")]
+    public async Task<List<Balance>?> GetBalances(
+        string network,
+        string address,
+        IStateService stateService)
+    {
+        Address agentAddress;
+        try
+        {
+            agentAddress = new Address(address);
+        }
+        catch (ArgumentException)
+        {
+            Response.StatusCode = StatusCodes.Status400BadRequest;
```

Comment:
How about apply this Address Type?
```csharp
// just example
public class ValidatedAddress
{
    public Address Address { get; private set; }
    public bool IsValid { get; private set; }
    public string ErrorMessage { get; private set; }

    public ValidatedAddress(string address)
    {
        try
        {
            Address = new Address(address);
            IsValid = true;
        }
        catch (ArgumentException ex)
        {
            IsValid = false;
            ErrorMessage = "The address provided is invalid. Please ensure it meets the formatting rules.";
        }
    }
}
```

usage
```csharp
    if (!validatedAddress.IsValid)
    {
        Response.StatusCode = StatusCodes.Status400BadRequest;
        return Json(new { error = validatedAddress.ErrorMessage });
    }
```

Reply:
@Atralupus Thanks for your inspiration! I pushed `AddressValidator` in 38f2564 and fa945ac. Please check it out.